### PR TITLE
prevent logging of reoccuring messages

### LINKED ISSUE #35: Don't log reoccurring errors / messages
reporter logs "in reporter lock" every ten seconds

### PR DESCRIPTION
**Pull request by tallywiesenberg**
added a filter and a filter handler to prevent the log from being flooded with repeat messages (annoying!)

the solution for now is to not send a log message if the last line of the log file contains the same message.

Closes #35

### CODE REVIEW

File: src/telliot_feed_examples/utils/log.py [hunk 1]
```diff
@@ -49,5 +60,6 @@ def get_logger(name: str) -> logging.Logger:
 
     logger.addHandler(output_file_handler)
     logger.addHandler(stdout_handler)
+    logger.addFilter(DuplicateFilter())
```
Comment (oraclown) [inline src/telliot_feed_examples/utils/log.py:63]: nice, very clean

File: tests/utils/test_logging.py [hunk 2]
```diff
@@ -24,3 +25,24 @@ def test_get_logger() -> None:
...
+    log_file = logger.handlers[0].baseFilename
+    expected_log_file = default_logsdir() / ("telliot-feed-examples.log")
+    expected_log_file.resolve().absolute()
+    num_lines_before = 0
+    with open(os.path.join(expected_log_file), "r") as f:
+        for line in f:
+            num_lines_before += 1
+
+    for _ in range(5):
+        logger.info("Address not yet staked. Depositing stake.")
```
Comment (oraclown) [inline tests/utils/test_logging.py:41]: Maybe change the message to something unique for the tests? That, or do this in a tests.log file

**Review by oraclown — APPROVED** [review]

File: tests/utils/test_logging.py (hunk 2 above)
Comment (tallywiesenberg) [reply tests/utils/test_logging.py:41]: Ok

File: tests/utils/test_logging.py [hunk 3]
```diff
@@ -24,3 +25,24 @@ def test_get_logger() -> None:
     assert isinstance(logger, Logger)
     assert log_file == str(expected_log_file)
     assert logger.name == __name__
+
+def test_reocurring_messages() -> None:
```
Comment (clamdad) [inline tests/utils/test_logging.py:29]: typo

Comment (tallywiesenberg) [reply tests/utils/test_logging.py:30]: oh yeah haha

Comment (tallywiesenberg) [reply tests/utils/test_logging.py:30]: i knew something looked off!

Comment (tallywiesenberg) [reply tests/utils/test_logging.py:30]: https://www.merriam-webster.com/words-at-play/difference-between-recur-and-reoccur-definitions or maybe not

**Pull request merged**